python/GafferSceneUI/VisualiserToolUI.py

```diff
@@ -118,3 +123,63 @@
 
 	},
 )
+
+def __setDataName( plug, name, checked ) :
+
+	plug.setValue( name )
+
+def __primitiveVariableContextMenu( menuDefinition, plugValueWidget ) :
+
+	plug = plugValueWidget.getPlug()
+	node = plug.node()
+	if not isinstance( node, GafferSceneUI.VisualiserTool ) :
+		return
+	if plug != node["dataName"] :
+		return
+
+	scenePlug = node.view()["in"].getInput()
+	scriptNode = scenePlug.ancestor( Gaffer.ScriptNode )
+	selection = GafferSceneUI.ScriptNodeAlgo.getSelectedPaths( scriptNode )
+
+	primVars = []
+
```

Comment:
We need to scope the View's context here before evaluating the scene at all. Here's a repro for a failure case :

- Focus the ContextVariables node.
- Pin the NameSwitch into the Viewer.
- Right-click and witness the lack of context variables.

<details>

```
import Gaffer
import GafferScene
import imath

Gaffer.Metadata.registerValue( parent, "serialiser:milestoneVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:majorVersion", 5, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:minorVersion", 1, persistent=False )
Gaffer.Metadata.registerValue( parent, "serialiser:patchVersion", 0, persistent=False )

__children = {}

__children["SceneReader"] = GafferScene.SceneReader( "SceneReader" )
parent.addChild( __children["SceneReader"] )
__children["SceneReader"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["Cube"] = GafferScene.Cube( "Cube" )
parent.addChild( __children["Cube"] )
__children["Cube"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["NameSwitch"] = Gaffer.NameSwitch( "NameSwitch" )
parent.addChild( __children["NameSwitch"] )
__children["NameSwitch"].setup( GafferScene.ScenePlug( "value", ) )
__children["NameSwitch"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["ContextVariables"] = Gaffer.ContextVariables( "ContextVariables" )
parent.addChild( __children["ContextVariables"] )
__children["ContextVariables"].setup( GafferScene.ScenePlug( "in", ) )
__children["ContextVariables"]["variables"].addChild( Gaffer.NameValuePlug( "", Gaffer.StringPlug( "value", defaultValue = '', flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ), True, "member1", Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic ) )
__children["ContextVariables"].addChild( Gaffer.V2fPlug( "__uiPosition", defaultValue = imath.V2f( 0, 0 ), flags = Gaffer.Plug.Flags.Default | Gaffer.Plug.Flags.Dynamic, ) )
__children["SceneReader"]["fileName"].setValue( '${GAFFER_ROOT}/resources/gafferBot/caches/gafferBot.scc' )
__children["SceneReader"]["__uiPosition"].setValue( imath.V2f( 23.1000023, 10.0999994 ) )
__children["Cube"]["__uiPosition"].setValue( imath.V2f( 4.70000458, 10.250001 ) )
__children["NameSwitch"]["__index"].setInput( __children["NameSwitch"]["__outIndex"] )
__children["NameSwitch"]["selector"].setValue( '${a}' )
__children["NameSwitch"]["in"][0]["name"].setValue( '*' )
__children["NameSwitch"]["in"][0]["value"].setInput( __children["Cube"]["out"] )
__children["NameSwitch"]["in"][1]["name"].setValue( 'A' )
__children["NameSwitch"]["in"][1]["value"].setInput( __children["SceneReader"]["out"] )
__children["NameSwitch"]["__uiPosition"].setValue( imath.V2f( 18.8000011, -2.06406093 ) )
__children["ContextVariables"]["variables"]["member1"]["name"].setValue( 'a' )
__children["ContextVariables"]["variables"]["member1"]["value"].setValue( 'A' )
__children["ContextVariables"]["in"].setInput( __children["NameSwitch"]["out"]["value"] )
__children["ContextVariables"]["__uiPosition"].setValue( imath.V2f( 18.8008385, -10.2281237 ) )


del __children

```
</details>